### DISCUSSION TITLE
Static check of partitions for MCUBoot?

### DISCUSSION THREAD

**Discussion by vChavezB** [zephyrproject-rtos/zephyr#76537] [Ideas]
I was wondering if the current state of MCUBoot allows to statically verify that the defined partitions (e.g. slot0, slot1) are compatible.

Currently if a user mistakenly gives less space to a partition, or flash layout is not compatible then MCUBoot will not do upgrades.  This can be checked if logging is enabled and sent an image (confirm) over mcumgr.

As far as I understand the dynamic check is done here (swap strategy)

https://github.com/zephyrproject-rtos/mcuboot/blob/fb2cf0ec3da3687b93f28e556ab682bdd4b85223/boot/bootutil/src/swap_move.c#L260

Would it be possible to do the same over cmake or python?

**Comment by vChavezB** [comment]
Just an update. I think it is not that easy as for some compatibility checks, the information of each sector is required. This is obtained via the page layout flash api. The page layout implementation is not defined in the device tree but in the driver of the flash memory. Hence this might not be practical.

**Reply by nordicjm** [reply]
If you build MCUboot with logging and MCUboot's logging set to debug it will output if the slots are compatible, it will also output if the slots are not optimally sized, this is a build debug log though not configure

**Reply by vChavezB** [reply]
Yes, through logging it is possible. My motivation is to reduce development time by having static checks and automatically detecting them through a CI system. This avoids having to flash a board and verify compatibility at runtime to approve a PR for an application with mcuboot. You could also have hardware attached to the CI and verify this, but this seems overkill.

The flash page layout typically does not change (although some flash memories can be changed at runtime). So the idea is basically to have static information about the flash layout. Unfortunately, I looked in the Flash API and mcuboot source and flash page layouts are not defined in the device tree.
